suite/runsuite.cmake

```diff
@@ -316,9 +316,12 @@ endif ()
 # to get the diff and check it.  The vera++ rules do check C/C++ code.
 
 # Check for trailing space.  This is a diff with an initial column for +-,
-# so a blank line will have one space: thus we rule that out.
+# so we need to exclude the following cases:
+# 1. existing blank line will now have one space;
+# 2. lines starting with -.
+# We do this by matching lines that start with + or a space and end with a space.
 # The clang-format check will now find these in C files, but not non-C files.
-string(REGEX MATCH "[^\n] \n" match "${diff_contents}")
+string(REGEX MATCH "^[+ ].* \n" match "${diff_contents}")
```

Comment:
It would be good to describe how this was tested in the PR description: even including a couple small examples, to give confidence it works (e.g., unless I'm missing something with CMake's ^ handling, I assume this would fail on many small tests: maybe all tests since there's usually a header line in the diff).

Reply:
I did test the regex in a line-based regex evaluator and thought the CMake `string(REGEX MATCH` is also line-based. I'll do more tests in CMake to make sure this is working, sorry.

Reply:
Unless the CMake regex has changed, I remember the ^ being fairly useless and having to check for \n all over the place in the regexes in our code.  https://groups.google.com/g/dynamorio-users/c/Yl5g5jM-NY8 says start of string, not start of line.

Reply:
Thanks, yes, `^` indeed matches the start of the string, but what surprises me the most is `.` also matches `\n`. :O 
 Anyway, I submitted a fix and updated the PR description with test cases, should be good now. I should be more careful with CMake stuff.